src/common/meta/src/state_store.rs

```diff
@@ -79,17 +80,21 @@ fn decode_kv(kv: KeyValue) -> Result<(String, Vec<u8>)> {
     Ok((key, value))
 }
 
-enum SplitValue<'a> {
-    Single(&'a [u8]),
-    Multiple(Vec<&'a [u8]>),
+enum SplitValue {
+    Single(Vec<u8>),
+    Multiple(Vec<Vec<u8>>),
 }
 
-fn split_value(value: &[u8], max_value_size: Option<usize>) -> SplitValue<'_> {
+fn split_value(value: Vec<u8>, max_value_size: Option<usize>) -> SplitValue {
     if let Some(max_value_size) = max_value_size {
         if value.len() <= max_value_size {
             SplitValue::Single(value)
         } else {
-            SplitValue::Multiple(value.chunks(max_value_size).collect::<Vec<_>>())
+            let mut values = vec![];
+            for chunk in value.into_iter().chunks(max_value_size).into_iter() {
```

Comment:
@WenyXu Will this introduce additional allocations? The original `SplitValue` holds `Vec<&[u8]>`.

Reply:
No. This call into_iter. Actually the previous code has an extra alloc. See LibraryLoader.

Reply:
We move the bytes here. Previously it uses a ref but we can actually take the ownership. This calls delegate to `itertools` `IntoChunks`, not the slice's `chunks` that may clone.